src/main/kotlin/nl/info/zac/app/zaak/ZaakKoppelenRestService.kt

```diff
@@ -127,18 +128,29 @@ class ZaakKoppelenRestService @Inject constructor(
 
     private fun Zaak.hasLinkRights() = policyService.readZaakRechten(this).koppelen
 
-    private fun ZaakZoekObject.isLinkableTo(sourceZaak: Zaak, relationType: RelatieType) =
+    private fun ZaakZoekObject.isLinkableTo(sourceZaak: Zaak, relationType: RelatieType): Boolean =
         when (relationType) {
-            RelatieType.HOOFDZAAK -> this.zaaktypeUuid?.let { uuid ->
-                ztcClientService.readZaaktype(sourceZaak.zaaktype).deelzaaktypen.any {
-                    it.toString().contains(uuid)
-                }
-            } ?: false
-            RelatieType.DEELZAAK -> sourceZaak.zaaktype.extractUuid().toString().let { uuid ->
-                ztcClientService.readZaaktype(UUID.fromString(this.zaaktypeUuid)).deelzaaktypen.any {
-                    it.toString().contains(uuid)
-                }
-            }
+            RelatieType.HOOFDZAAK ->
+                // hoofdzaak to hoofdzaak link not allowed
+                !sourceZaak.is_Hoofdzaak &&
+                    // a zaak cannot have two hoofdzaken
+                    !sourceZaak.isDeelzaak &&
+                    // source zaak's zaaktype is allowed in target zaak as deelzaak
+                    sourceZaak.zaaktype.extractUuid().toString().let { uuid ->
+                        ztcClientService.readZaaktype(UUID.fromString(this.zaaktypeUuid)).deelzaaktypen.any {
+                            it.toString().contains(uuid)
+                        }
+                    }
+            RelatieType.DEELZAAK ->
+                // As per https://vng-realisatie.github.io/gemma-zaken/standaard/zaken
+                // "deelzaken van deelzaken zijn NIET toegestaan"
+                !sourceZaak.isDeelzaak &&
+                    // target zaak's zaaktype is allowed in source zaak as deelzaak
+                    this.zaaktypeUuid?.let { uuid ->
+                        ztcClientService.readZaaktype(sourceZaak.zaaktype).deelzaaktypen.any {
+                            it.toString().contains(uuid)
+                        }
+                    } ?: false
             else -> throw UnsupportedOperationException("Unsupported link type: $relationType")
```

Comment:
You loose a lot of information here. I think you also want to include the source and target zaak UUIDs here at least. Otherwise troubleshooting will be next to impossible.

Reply:
Added short `source -> target` ids